pkg/daemon/ceph/osd/kms/azure.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2024 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kms
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/libopenstorage/secrets"
+	"github.com/libopenstorage/secrets/azure"
+	"github.com/pkg/errors"
+	"github.com/rook/rook/pkg/clusterd"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	//#nosec G101 -- This is only the k8s secret name
+	azureClientCertSecretName = "AZURE_CERT_SECRET_NAME"
+)
+
+// IsAzure determines whether the configured KMS is Azure Key Vault
+func (c *Config) IsAzure() bool {
+	return c.Provider == secrets.TypeAzure
+}
+
+// InitAzure initializes azure key vault client
+func InitAzure(ctx context.Context, context *clusterd.Context, namespace string, config map[string]string) (secrets.Secrets, error) {
+
```

Comment:
nit: remove blank line at start of method

Reply:
done.